### PR TITLE
Fixed issue #838

### PR DESCRIPTION
**Pull request by markroxor**
Fixed RuntimeWarning: overflow encountered in exp.

### CODE REVIEW

File: gensim/models/word2vec.py [hunk 1]
```diff
@@ -273,7 +273,14 @@ def train_sg_pair(model, word, context_index, alpha, learn_vectors=True, learn_h
         l1 += neu1e * lock_factor  # learn input -> hidden (mutates model.syn0[word2.index], if that is l1)
     return neu1e
 
-
+def sigmoid(p):
+    if p > 0:
+        return 1. / (1. + exp(-p))
+    elif p <= 0:
+        exp(p) / (1 + exp(p))
```
Comment (tmylk) [inline gensim/models/word2vec.py]: Should it say return?

Comment (markroxor) [inline gensim/models/word2vec.py]: That's embarrassing, forgot to return in all the haste. Sending a new PR.

**Comment by tmylk** [comment]
Please add a note in changelog as well

**Comment by markroxor** [comment]
Should I add a new version as well or continue with 0.13.3?

**Comment by tmylk** [comment]
0.13.3 is the next release. Please add to it. Thanks

**Comment by tmylk** [comment]
Thanks for the report and the fix!

**Comment by markroxor** [comment]
Didn't put much of an effort but thanks @tmylk :)

File: gensim/models/word2vec.py [hunk 2]
```diff
@@ -273,7 +273,14 @@ def train_sg_pair(model, word, context_index, alpha, learn_vectors=True, learn_h
         l1 += neu1e * lock_factor  # learn input -> hidden (mutates model.syn0[word2.index], if that is l1)
     return neu1e
 
-
+def sigmoid(p):
+    if p > 0:
+        return 1. / (1. + exp(-p))
+    elif p <= 0:
+        return exp(p) / (1 + exp(p))
+    else:
```
Comment (piskvorky) [inline gensim/models/word2vec.py]: What is this entire `if` for?

The logic seems obscure (why branch at all?) and weird (what is the `else` for?). @tmylk

Comment (tmylk) [inline gensim/models/word2vec.py]: @markroxor Please remove else and add comment explaining #838

**Pull request merged**